command/agent/auth/azure/azure.go

```diff
@@ -73,7 +72,7 @@ func NewAzureAuthMethod(conf *auth.AuthConfig) (auth.AuthMethod, error) {
 	return a, nil
 }
 
-func (a *azureMethod) Authenticate(ctx context.Context, client *api.Client) (retPath string, header http.Header, retData map[string]interface{}, retErr error) {
+func (a *azureMethod) Authenticate(ctx context.Context) (retPath string, header http.Header, retData map[string]interface{}, retErr error) {
```

Comment:
I think this needs the client parameter to satisfy the [AuthMethod interface](https://github.com/hashicorp/vault/blob/74a613545e2e1baef8517e30ab5595e5ab4da1fc/command/agent/auth/auth.go#L22-L31), which is returned from NewAzureAuthMethod() above.

But since it's not used in this function, maybe just use a blank identifier?
```suggestion
func (a *azureMethod) Authenticate(ctx context.Context, _ *api.Client) (retPath string, header http.Header, retData map[string]interface{}, retErr error) {
```